### PR TITLE
Allow non-admin users to delete review requests

### LINKED ISSUE #14459: Delete review requests as non-admin
- Gitea version (or commit ref): 1.14
- Can you reproduce the bug at https://try.gitea.io:
  - [X] Yes 

## Description

When I request a review, I am not able to delete this review request if I am no administrator. I would expect at least to be able to delete a review request I created myself in case I selected the wrong person by accident. But I think even someone else should be able to delete the review request as well. Everyone that can add reviewers should also be able to delete them in my opinion?

## Screenshots

I add a review request:

image

But I cannot delete it:

image

Unless when I am admin:

image

### PR DESCRIPTION
**Pull request by Zettat123**
Fix #14459

The following users can add/remove review requests of a PR
- the poster of the PR
- the owner or collaborators of the repository
- members with read permission on the pull requests unit

### CODE REVIEW

File: models/fixtures/email_address.yml [hunk 1]
```diff
@@ -293,3 +293,35 @@
...
+  is_activated: true
+  is_primary: true
+
+-
+  id: 41
+  uid: 41
+  email: org41@example.com
+  lower_email: org41@example.com
+  is_activated: true
+  is_primary: true
```
Comment (KN4CK3R) [inline models/fixtures/email_address.yml]: An organization does not have a `email_address` entry.

**Review by lunny — APPROVED** [review]

**Review by wolfogre — APPROVED** [review]

**Comment by wolfogre** [comment]
BTW, I think it's a design change (even somewhat breaking) instead of bug fixing. I'm not sure if it's a good idea to backport it.

**Comment by Zettat123** [comment]
> > members with read permission on the pull requests unit
> 
> Erm… `read`? Why not `write`?

In my view, a user who can read the PR is also likely to know who should/shouldn't review the PR. Also, I followed @lunny 's idea in https://github.com/go-gitea/gitea/issues/14459#issuecomment-1923749566 where the `read` permission was mentioned. So I only check the `read` permission here.

**Comment by lunny** [comment]
> > members with read permission on the pull requests unit
> 
> 
> 
> Erm… `read`?
> 
> Why not `write`?

Read permission of issue or pr units can create issue or pr.

**Comment by silverwind** [comment]
Ready for merge?

**Pull request merged**